components/fonts/font.rs

```diff
@@ -365,10 +396,30 @@ pub struct ShapingOptions {
     pub word_spacing: Au,
     /// The Unicode script property of the characters in this run.
     pub script: Script,
+    /// Set of variations provided by the `font-variation-settings` property.
+    pub variation_settings: Vec<(Tag, VariationValue)>,
     /// Various flags.
     pub flags: ShapingFlags,
 }
 
+/// Wrapper around f32 to provide `Hash` and `Eq` impls.
+///
+/// `f32` doesn't implement these traits because floating point precision would
+/// make them unreliable. In our case, we know two things:
+/// * There are likely no calculations done on the value (except for `calc` in `font-variation-settings`)
+/// * The trait impls are only used as an index into the shaping cache, so imprecisions will only cause cache misses.
+///
+/// We also need to use a `f32` because both stylo and harfbuzz use a floating point value.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct VariationValue(pub f32);
+
+impl Eq for VariationValue {}
+impl Hash for VariationValue {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        state.write_u32(self.0.to_bits());
+    }
+}
+
```

Comment:
Instead of implementing a new type with the same workaround for `Hash`, let's just pass around `webrender_api::FontVariation` everywhere.